### PR TITLE
Feature/#190 file logger

### PR DESCRIPTION
**Pull request by timaschew**
proposal for squashed commit message:

**introduce logger, refactor require lookup and logger initialization**
- replace std-out-logger with winston
  - which can be configured via config.yml
  - and supports multiple transports in parallel
  - still possible to use own logger with previous deepstream logger API
- refactor initialization of logger during deepstream start, since logger should be loaded before any plugins are loaded
- replace  utils.normalisePath with utils.lookupRequirePath due to handling all cases

### CODE REVIEW

File: src/deepstream.io.js [hunk 1]
```diff
@@ -109,12 +109,17 @@ Deepstream.prototype.set = function( key, value ) {
  * @returns {void}
  */
 Deepstream.prototype.start = function() {
```
Comment (yasserf) [inline src/deepstream.io.js]: Update documentation to reflect logger initializing first

File: src/deepstream.io.js [hunk 2]
```diff
@@ -109,12 +109,17 @@ Deepstream.prototype.set = function( key, value ) {
...
+	if( !this.loggerInitializer ) {
+		this.loggerInitializer = new DependencyInitialiser( this._options, 'logger' );
+		this.loggerInitializer.once( 'ready',
+			this._checkReady.bind( this, 'logger', this.loggerInitializer.getDependency() )
+		);
+		this.loggerInitializer.once( 'ready', this._start.bind( this ) );
 	}
+};
+
+Deepstream.prototype._start = function() {
```
Comment (yasserf) [inline src/deepstream.io.js]: documentation

File: src/utils/config-initialiser.js [hunk 3]
```diff
@@ -59,9 +58,22 @@ function handleUUIDProperty( config ) {
  * @private
  * @returns {void}
  */
-function handleLogLevel( config ) {
+function handleLogger( config ) {
+	let configClazz;
```
Comment (yasserf) [inline src/utils/config-initialiser.js]: class

File: conf/config.yml [hunk 4]
```diff
@@ -21,28 +21,13 @@ sslCa: null
...
-#       level: debug
-#       options:
-#         filename: 'logs.json'
-#     -
-#       type: time
-#       level: warn
-#       options:
-#         filename: time-roated-logfile
-#         datePattern: .yyyy-MM-dd-HH-mm
+#         colorize: true
```
Comment (yasserf) [inline conf/config.yml]: is colorize winston naming convention?

Comment (timaschew) [inline conf/config.yml]: yes

File: src/deepstream.io.js [hunk 5]
```diff
@@ -181,21 +216,25 @@ Deepstream.prototype.convertTyped = function( value ) {
 };
 
 /**
- * Synchronously loads a configuration file and returns
- * the result
+ * Synchronously loads a configuration file
+ * Initialization of plugins and logger will be triggered by the
+ * configInitialiser, but it should not block. Instead the ready events of
+ * those plugins are handled through the DependencyInitialiser in this instnace.
```
Comment (yasserf) [inline src/deepstream.io.js]: Typo

**Pull request merged**